### ISSUE TITLE
OCR: Improving accuracy and evaluating continuation

### ISSUE THREAD

**Issue by aerugo** [opencollective/opencollective#7271]
Since we launched the OCR feature, the general feedback has been that it's impressive when it works and annoying when it doesn't. It saves our users time and effort, but accuracy still needs to be improved. We have worked with Klippa to improve the accuracy, but we could do better with a few tweaks.

![Image](https://github.com/opencollective/opencollective/assets/7785081/28d9bcd9-269f-4b87-a988-647796af124b)

This causes friction because if OCR accuracy is lower than the accuracy of a non-assisted human, it causes extra work for fiscal host admins. We think we can significantly improve accuracy by combining the OCR results with our knowledge about the context. For example, we could require the model's confidence to be particularly high to suggest a currency other than the collective's currency.

In this project, an engineer and a product manager would work together to come up with several such tweaks, evaluate them against production data, and implement them if they are proven to work to improve accuracy.

While evaluating the model, we should also take the opportunity to research alternatives to the current solution. Given that the results are not as good as we had hoped out of the box, we should consider if we could train a custom model. Our platform processes an impressive amount of fiscal documents, and that number is likely to grow. We are in a good position to train our model, and there are pre-trained open-source neural networks available, which means we would not need to start from scratch. Our contract with the current supplier started on September 1st, 2023, and is valid for 12 months before renewal. Now is a good time to evaluate building the technology in-house. Making a first assessment of that possibility would also be a part of this project.

## Objectives

- [ ] Improve OCR accuracy by coming up with, evaluating and implementing tweaks that rely on our contextual domain knowledge
- [ ] Evaluating the feasibility of switching to our own model after the current contract runs out in September

**Comment by Betree** [comment]
That's a good one to fit into the coming sprint. The 1st part should be pretty straightforward.

**Comment by Betree** [comment]
About "Invoice number", when uploading a custom doc:
> it's a place where machine detection of the invoice number would shine.

See https://oficonsortium.slack.com/archives/C05KG5KB80P/p1735917880273509?thread_ts=1735912379.156219&cid=C05KG5KB80P

**Comment by Betree** [comment]
OCR with Klippa has been disabled for now. We'll beck how to bring that back with https://github.com/opencollective/opencollective/issues/8094.
